core/src/main/java/com/linecorp/armeria/client/retry/RetryingRpcClient.java

```diff
@@ -114,9 +114,14 @@ private void doExecute0(ClientRequestContext ctx, RpcRequest req, DefaultRpcResp
                 }
             } else {
                 response.handle(voidFunction((result, thrown) -> {
-                    if (result != null) {
+                    // !!! DO NOT use "result != null" as the success/fail check condition !!!
```

Comment:
IMO we can remove these comments since handle/whenComplete is a common pattern in the codebase and we wouldn't add this to every single call. Would be very cool if it's possible to make a checkstyle rule for this though.

With the new unit test, there shouldn't be a regression either way, thanks :)

Reply:
comments deleted